runtime/compiler/x/codegen/J9TreeEvaluator.cpp

```diff
@@ -12581,7 +12581,7 @@ J9::X86::TreeEvaluator::generateTestAndReportFieldWatchInstructions(TR::CodeGene
    TR::MemoryReference *classFlagsMemRef = NULL;
    TR_J9VMBase *fej9 = (TR_J9VMBase *)(cg->fe());
    bool isInstanceField = node->getOpCode().isIndirect();
-   bool isAOTCompile = cg->comp()->compileRelocatableCode();
+   bool isAOTCompile = cg->comp()->compileRelocatableCode() || cg->comp()->isOutOfProcessCompilation();
```

Comment:
Do we need similar changes on Power and Z?

Reply:
Yes I think we do. I will add them in.

Reply:
I'm not fully ready with this change just yet (also description hasn't been added), I'll ping you once I'm ready.